cocotb/regression.py

```diff
@@ -316,7 +316,12 @@ def _result_was():
                 return
 
         elif test.expect_error:
-            self.log.info("Test errored as expected: " + _result_was())
+            if test.expect_error is True or isinstance(result, test.expect_error):
```

Comment:
Could simplify this logic perhaps with a patch like:
```diff
--- C:\Users\wiese\Repos\cocotb\cocotb\decorators.py
+++ C:\Users\wiese\Repos\cocotb\cocotb\decorators.py
@@ -437,7 +437,7 @@
             value representing simulation timeout (not implemented).
         expect_fail (bool, optional):
             Don't mark the result as a failure if the test fails.
-        expect_error (bool, optional):
+        expect_error (bool, Type[Exception], optional):
             Don't mark the result as an error if an error is raised.
             This is for cocotb internal regression use 
             when a simulator error is expected.
@@ -450,6 +450,10 @@
                  skip=False, stage=None):
         super(test, self).__init__(f)
 
+        if expect_error is True:
+            expect_error = (Exception,)
+        elif expect_error is False:
+            expect_error = ()
         self.timeout = timeout
         self.expect_fail = expect_fail
         self.expect_error = expect_error

```

This saves you the `is True` comparison at the call site, and lets you use `isinstance(result, test.expect_error)` unconditionally

Reply:
Good idea, will capture the docstring changes as well.